File: frontend/src/services/laboratoryService.js
import apiClient from './apiClient';

const laboratoryService = {
  async getAll() {
    const { data } = await apiClient.get('/laboratories');
    return data.data;
  },
  async create(payload) {
    const { data } = await apiClient.post('/laboratories', payload);
    return data.data;
  },
  async update(id, payload) {
    const { data } = await apiClient.put(`/laboratories/${id}`, payload);
    return data.data;
  },
  async remove(id) {
    await apiClient.delete(`/laboratories/${id}`);
  },
};

export default laboratoryService;

File: backend/src/main/java/com/hospital/hms/service/LaboratoryService.java
package com.hospital.hms.service;

import com.hospital.hms.dto.request.LaboratoryRequest;
import com.hospital.hms.dto.response.LaboratoryResponse;

import java.util.List;

public interface LaboratoryService {

    LaboratoryResponse create(LaboratoryRequest request);

    LaboratoryResponse update(Long id, LaboratoryRequest request);

    LaboratoryResponse getById(Long id);

    List<LaboratoryResponse> getAll();

    void delete(Long id);
}
